ipaclient/plugins/vault.py

```diff
@@ -969,7 +970,7 @@ def forward(self, *args, **options):
         json_vault_data = decoding_ctx.cipher_op(wrapped_vault_data)\
             + decoding_ctx.digest_final()
 
-        vault_data = json.loads(json_vault_data)
+        vault_data = json.loads(decode_json(json_vault_data))
```

Comment:
It looks like `json_vault_data` is always an instance of `bytes`, so instead of guessing the encoding, we should always decode it as UTF-8. Ditto for encoding `json_vault_data` in `vault_archive.forward()`.